### DISCUSSION TITLE
Badge System

### DISCUSSION THREAD

**Discussion by alesanchezr** [breatheco-de/breatheco-de#2577] [General]
We need a way to represent user skills inside of breathecode, I'm proposing using a Badge System.

## Badge System

- A `badge` is something you earn (like a medal), it represent skills that you have. For example: `html badge`.
- You can earn and collect `skill points` from completing projects, practices and quizzes.
- To earn a badge you have to accumulate enough `skill points` until you hit the minimum amount required to "earn it".
- [This is a proposal](https://github.com/breatheco-de/apiv2/blob/master/breathecode/certificate/models.py#L47) for the table badge.
- Specialties contain multiple badges, if you collect all the badges from a specialty, you become an expert on that specialty.
- bc.certificates contain Specialties

**Comment by jefer94** [comment]
Is possible create many signals to add a badge to a user using three models

```
ExternalReference
- path (ex: admissions.Cohort)
- reference_id (not linked)
```

```
Badge
- name
- slug (unique)
- description
- external_references (to many)
- created_at
- updated_at
```

```
UserBadge
- user
- Badge
- created_at
```

Through of UserSpecialty, Task and Cohort you should build the first badges with a script, but this tasks just can be used by breathecode, for the new 4geeks system this system not have nothing to do yet
